code/cvss/src/Security/CVSS.hs

```diff
@@ -0,0 +1,289 @@
+{-# LANGUAGE ImportQualifiedPost #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE PatternSynonyms #-}
+
+{- | This module provides a CVSS parser and utility functions
+ adapted from https://www.first.org/cvss/v3.1/specification-document
+-}
+module Security.CVSS (
+    -- * Type
+    CVSS,
```

Comment:
I think hiding the data constructor is fine.  Essentially really, to avoid construction of invalid data.  But we *should* export `cvssVersion` and the `CVSSVersion` type; these will be important in many use cases.

And we should also have a function to convert a `CVSS` to a CVSS vector string.  It would be good to have two variants: one that re-serialises the vector in the original order, and one that serialises the vector in the preferred order (see e.g. CVSS 3.1 spec Chapter 6).